model/riscv_pmp_control.sail

```diff
@@ -104,15 +104,41 @@ function pmpCheckRWX(ent, acc) = {
   }
 }
 
-// this needs to be called with the effective current privilege.
-val pmpCheckPerms: (Pmpcfg_ent, AccessType(ext_access_type), Privilege) -> bool
+/* this needs to be called with the effective current privilege */
+val pmpCheckPerms: (Pmpcfg_ent, AccessType(ext_access_type), Privilege) -> bool effect {rreg}
 function pmpCheckPerms(ent, acc, priv) = {
+  if mseccfg.MML() == 0b1 then {
+    if ent.R() == 0b0 & ent.W() == 0b1 then {
+      if pmpLockBit(ent) then {
+        match acc {
+          Execute(_) => true,
+          Read(_)    => priv == Machine & ent.X() == 0b1,
+          _          => false
+        }
+      } else {
+      match acc {
```

Comment:
*Please* read your own PR and fix all the indentation issues, many are not properly nested. It's tiring going through and saying the same thing over and over.